### PR TITLE
update dependency RFC links

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

Updates RFC links in the dependency management RFC suite.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: text/dependencies/rfcs/0003-dependency-management-overview.md
Comment (ryanmoran) [inline text/dependencies/rfcs/0003-dependency-management-overview.md:124]: You can always use relative paths for these links so that you don't have to mention the branch name at all.

```suggestion
1. Federated Model Adoption In Buildpacks: [Phase 1 RFC](./text/dependencies/rfcs/0004-dependency-management-phase-one.md)
```

**Review by ekcasey — APPROVED** [review]

**Comment by ekcasey** [comment]
Looks like this needs a review from one of the @paketo-buildpacks/dependencies-maintainers to merge.

**Review by brayanhenao — APPROVED** [review]

**Comment by joshuatcasey** [comment]
2022-08-16 Working Group Notes: just needs to be merged

**Pull request merged**
